src/Functions/URL/resolveRelativeURL.cpp

```diff
@@ -0,0 +1,218 @@
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument of function {}", arguments[0]->getName(), getName());
+
+        if (!isString(arguments[1]) && !isArray(arguments[1]))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument of function {}", arguments[1]->getName(), getName());
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t /*input_rows_count*/) const override
+    {
+        const ColumnPtr relative_url_column = arguments[0].column;
+        const ColumnPtr base_url_column = arguments[1].column;
+
+        const ColumnString * relative_url = checkAndGetColumn<ColumnString>(relative_url_column.get());
+
+        const auto * base_url = checkAndGetColumnConstData<ColumnString>(base_url_column.get());
+
+        if (!relative_url)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN, "Illegal column {} of argument of function {}", arguments[0].column->getName(), getName());
+        }
+
+        if (!base_url)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN, "Illegal column {} of argument of function {}", arguments[1].column->getName(), getName());
+        }
+
+        auto col_res = ColumnString::create();
+
+        ColumnString::Chars & vec_res = col_res->getChars();
+        ColumnString::Offsets & offsets_res = col_res->getOffsets();
+        vector(relative_url->getChars(), relative_url->getOffsets(), base_url->getChars(), base_url->getOffsets(), vec_res, offsets_res);
+        return col_res;
+    }
+
+    static void vector(
+        const ColumnString::Chars & relative_url_data,
+        const ColumnString::Offsets & relative_url_offsets,
+        const ColumnString::Chars & base_url_data,
+        const ColumnString::Offsets & base_url_offsets,
+        ColumnString::Chars & res_data,
+        ColumnString::Offsets & res_offsets)
+    {
+        res_data.reserve(relative_url_data.size() / 5);
+        res_offsets.resize(relative_url_offsets.size());
+
+        ColumnString::Offset relative_url_prev_offset = 0;
+        ColumnString::Offset base_url_prev_offset = 0;
+        ColumnString::Offset res_offset = 0;
+
+        for (size_t i = 0; i < relative_url_offsets.size(); ++i)
+        {
+            ColumnString::Offset relative_url_offset = relative_url_offsets[i];
+            ColumnString::Offset base_url_offset = base_url_offsets[i];
+
+
+            const char * relative_url_begin = reinterpret_cast<const char *>(&relative_url_data[relative_url_prev_offset]);
+            const char * relative_url_end = reinterpret_cast<const char *>(&relative_url_data[relative_url_offset]);
+            size_t relative_size = relative_url_end - relative_url_begin - 1;
+            char * relative_url_ptr = const_cast<char *>(relative_url_begin);
```

Comment:
Why do we need `const_cast` ? If you need to move `relative_url_ptr` with increment/decrement `const char *` should suit, because it guarantee data constness, but not variable itself. Data should remain to be constant